src/Microsoft.CodeQuality.Analyzers/UnitTests/QualityGuidelines/AvoidDuplicateElementInitializationTests.cs

```diff
@@ -0,0 +1,282 @@
+
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeQuality.CSharp.Analyzers.QualityGuidelines;
+using Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeQuality.Analyzers.QualityGuidelines.UnitTests
+{
+    public partial class AvoidDuplicateElementInitializationTests : DiagnosticAnalyzerTestBase
+    {
+        protected override DiagnosticAnalyzer GetBasicDiagnosticAnalyzer()
+        {
+            return new AvoidDuplicateElementInitialization();
+        }
+
+        protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
+        {
+            return new AvoidDuplicateElementInitialization();
+        }
+
+        [Fact]
+        public void NoInitializer()
+        {
+            VerifyCSharp(@"
+class C
+{
+    void Foo()
+    {
+        var x = new System.Collections.Generic.Dictionary<int, string>();
+    }
+}
+");
+        }
+
+        [Fact]
+        public void LiteralIntIndex()
+        {
+            VerifyCSharp(@"
+class C
+{
+    void Foo()
+    {
+        var x = new System.Collections.Generic.Dictionary<int, string>
+        {
+            [1] = ""a"",
+            [2] = ""b"",
+            [1] = ""c""
+        };
+    }
+}
+",
+                GetCSharpResultAt(10, 13, "1"));
+        }
+
+        [Fact]
+        public void CalculatedIntIndex()
+        {
+            VerifyCSharp(@"
+class C
+{
+    void Foo()
+    {
+        var x = new System.Collections.Generic.Dictionary<int, string>
+        {
+            [1] = ""a"",
+            [2] = ""b"",
+            [(10 + 3) * 2 - 25] = ""c""
+        };
+    }
+}
+",
+                GetCSharpResultAt(10, 13, "1"));
+        }
+
+        [Fact]
+        public void LiteralStringIndex()
+        {
+            VerifyCSharp(@"
+class C
+{
+    void Foo()
+    {
+        var dictionary = new System.Collections.Generic.Dictionary<string, int>
+        {
+            [""a""] = 1,
+            [""b""] = 2,
+            [""a""] = 3
+        };
+    }
+}
+",
+                GetCSharpResultAt(10, 13, "a"));
+        }
+
+        [Fact]
+        public void ConcatenatedStringIndex()
+        {
+            VerifyCSharp(@"
+class C
+{
+    void Foo()
+    {
+        var x = new System.Collections.Generic.Dictionary<string, int>
+        {
+            [""ab""] = 1,
+            [""bc""] = 2,
+            [""a"" + ""b""] = 3
+        };
+    }
+}
+",
+                GetCSharpResultAt(10, 13, "ab"));
+        }
+
+        [Fact]
+        public void EnumIndex()
+        {
+            VerifyCSharp(@"
+class C
+{
+    void Foo()
+    {
+        var dictionary = new System.Collections.Generic.Dictionary<System.DateTimeKind, int>
+        {
+            [System.DateTimeKind.Local] = 1,
+            [System.DateTimeKind.Utc] = 2,
+            [System.DateTimeKind.Local] = 3
+        };
+    }
+}
+",
+                // TODO: See if there's a way to use 'DateTimeKind.Local' here.
+                GetCSharpResultAt(10, 13, "2"));
```

Comment:
DateTimeKind is just an enum, so you should be able to cast it to an int, then to a string just fine.

Reply:
The constant gives the underlying int value, which is then converted to a string in the error message. I would like the error message to use the enum member name instead though ("DateTimeKind.Local" in this case) since that is likely how the user specified the index in the original code, so this is what the TODO is about.